src/main/openapi/discovery.yaml

```diff
@@ -209,15 +209,58 @@ components:
     DatasetSearchQueryFacet:
       type: object
       properties:
-        facetGroup:
+        source:
           type: string
-          title: Facet group
-        facet:
+          title: Facet source
+        type:
+          $ref: "#/components/schemas/FilterType"
+          title: Facet type
+        key:
+          type: string
+          title: Facet key
+        value:
+          type: string
+          title: Facet value
+        operator:
+          $ref: "#/components/schemas/Operator"
+          title: Operator
+        entries:
+          type: array
+          items:
+            $ref: '#/components/schemas/QueryEntry'
+      required:
+        - source
+        - type
+        - label
+    FilterType:
+      type: string
+      enum:
+        - DROPDOWN
+        - FREE_TEXT
+        - ENTRIES
+      title: Filter type
+    Operator:
+      type: string
+      enum:
+        - "="
+        - ">"
+        - "<"
+        - "!="
+        - "%"
+      x-enum-varnames:
```

Comment:
question: is this part of OpenAPI specification? How does the java generator understand this?

Reply:
OpenAPI generator supports it for OpenAPI 2.0 and 3.0. 

Varnames are the enum members and their values are set to the operators.